lib/ansible/modules/network/nxos/nxos_vrf.py

```diff
@@ -266,7 +266,7 @@ def map_obj_to_commands(updates, module):
                     commands.append('shutdown')
                 commands.append('exit')
 
-                if interfaces:
+                if interfaces and interfaces[0] != 'default':
```

Comment:
Why check only the first item of the list to be `default`?

Reply:
interfaces is a list parameter. It allows an array of interfaces or 'default' keyword (to remove all the existing interfaces on the device). Since it is list, I am checking the first element to see if it is 'default' or not. 
I followed this example:
https://github.com/ansible/ansible/blob/3a144b290d6161ced77df5c8912dad3af646197b/lib/ansible/modules/network/nxos/nxos_vxlan_vtep_vni.py#L326

If there some other way, let me know.